# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: aeacd48f565e
# Full Hash: aeacd48f565e73a85fff14ef1b109c5964437521
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
#   
# ==============================================================================

diff -r a5cd8e8ce4b5 -r aeacd48f565e js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Tue Nov 03 10:20:49 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Tue Nov 03 10:22:41 2020 +0000
@@ -4416,6 +4416,9 @@
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
 
+wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp);
+wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp);
+
 }  // namespace jit
 }  // namespace js
 